### PR TITLE
Office invoice flow [delivers #161689746]

### PR DESCRIPTION
**Pull request by stangah**
## Description

Kind of a lot going on here, sorry:
- Adds `GetShipmentInvoices` endpoint, retrieves all invoices with a matching `ShipmentID`
- Pulls code into a `DeliverAndPriceShipment` service, so that action can be performed on a test shipment when standing up data using `db_populate_e2e`
- Makes sure our test data doesn't have dates that fall on a weekend (this can blow up the rate engine I think)
- Adds `Invoice` to our normalized entity schema
- Adds selectors to fetch shipmentLineItems for a given invoice ID
- Refactors some of our invoice components a bit:
  - `InvoicePanel` still does most of the orchestration
  - `LineItemTable` just shows a table of line items and a total
  - `UnbilledTable` has both the "Approve Payment" widget and a `LineItemTable` of unbilled line items
  - `InvoiceTable` shows invoice info and a `LineItemTable` of line items for that invoice
  - `InvoicePayment` uses both internal component state and `entities` methods to manage UI

## Setup
Part of this PR is making sure that the move with locator `"DOOB"` (not my name :p) has all the expected `ShipmentLineItems` priced and ready to go to be invoiced, so test on that. You'll need ot re-run:

```sh
make db_populate_e2e
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](./docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](./docs/backend.md#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](./docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/161689746) for this change

## Screenshots

screen shot 2018-12-12 at 10 26 08 pm

### CODE REVIEW

File: src/shared/Invoice/InvoicePayment.jsx [hunk 1]
```diff
@@ -10,60 +10,98 @@ export const PAYMENT_APPROVED = 'IN_APPROVED_FLOW';
 export const PAYMENT_FAILED = 'IN_FAILURE_FLOW';
```
Comment (rdhariwal) [inline src/shared/Invoice/InvoicePayment.jsx]: we won't need these constants anymore :)

File: src/shared/Invoice/InvoicePayment.jsx [hunk 2]
```diff
@@ -10,60 +10,98 @@ export const PAYMENT_APPROVED = 'IN_APPROVED_FLOW';
...
-        break;
+        </div>
+      );
     }
-    return paymentContainer;
+
+    return (
+      <div>
+        {paymentAlert}
+        {header}
```
Comment (rdhariwal) [inline src/shared/Invoice/InvoicePayment.jsx]: should `paymentAlert` and `header` be rolled into one block so it is easier to reason? I think using redux and internal state is making that choice harder for us. This comment is more food for thought

File: src/shared/Invoice/UnbilledTable.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          createInvoiceStatus={this.props.createInvoiceStatus}
+        />
+        {itemsComponent}
+      </div>
+    );
+  }
+}
+
+UnbilledTable.propTypes = {
+  isInvoiceInDraft: PropTypes.bool,
```
Comment (rdhariwal) [inline src/shared/Invoice/UnbilledTable.jsx]: looks like unused prop, kill it with fire?

**Comment by tinyels** [comment]
This looks great! Let's create a chore for adding more tests and cleaning up the dead code.

**Pull request merged**